sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+
+                    new_labels = np.array([mapper[j] for j in self.labels])
+
+                    labels = labels[0:len(labels) - right_overlap]
+                    labels = np.concatenate((labels, new_labels))
+
+                right_overlap = len(X[np.isin(X[:, 0],
+                                              period[-frame_overlap + 1:])])
+
+        self.labels = labels
+        return self
+
+    def _predict(self, X):
+        """Predict the closest cluster each sample in X belongs to.
+
+        Parameters
+        ----------
+        X : data to cluster based on model formed in _fit, of type self.get_tag(
+        "X_inner_mtype")
+        y: ignored, exists for API consistency reasons.
+
+        Returns
+        -------
+        np.ndarray (1d array of shape (n_instances,))
+            Index of the cluster each time series in X belongs to.
+        """
+
+        X = check_array(X)
+
+        dist_matrix = squareform(pdist(X, metric=self.dist))
+        predicted_labels = self.model_.fit_predict(dist_matrix)
+
+        return predicted_labels
+
+    def _get_fitted_params(self):
+        """Get fitted parameters.
+
+        private _get_fitted_params, called from get_fitted_params
+
+        State required:
+            Requires state to be "fitted".
+
+        Returns
+        -------
+        fitted_params : dict with str keys
+            fitted parameters, keyed by names of fitted parameter
+        """
+
+        fitted_params = {
+        "eps1": self.eps1,
+        "eps2": self.eps2,
+        "min_samples": self.min_samples,
+        "metric": self.metric,
+        "n_jobs": self.n_jobs,
+        "algorithm": self.algorithm,
+        "leaf_size": self.leaf_size,
+        "metric_params": self.metric_params,
+        "p": self.p,
+        "dist": self.dist,
+        "labels_": self.labels_,
+        "dbscan__core_sample_indices_": self.model_.core_sample_indices_,
+        "dbscan__components_": self.model_.components_,
+        "dbscan__n_features_in_": self.model_.n_features_in_
+        }
+
+        return fitted_params
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for clusterers.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+
```

Comment:
I would also suggest to add a minimal, e.g., empty parameter set, to test the defaults.